packages/flutter/test/gestures/gesture_binding_test.dart

```diff
@@ -175,15 +175,15 @@ void main() {
     expect(events[1], isA<PointerScrollEvent>());
   });
 
-  test('Should synthesize kPrimaryButton for touch', () {
+  test('Should synthesize kPrimaryButton for touch when no button is set', () {
     final Offset location = const Offset(10.0, 10.0) * ui.window.devicePixelRatio;
     const PointerDeviceKind kind = PointerDeviceKind.touch;
     final ui.PointerDataPacket packet = ui.PointerDataPacket(
       data: <ui.PointerData>[
         ui.PointerData(change: ui.PointerChange.add, kind: kind, physicalX: location.dx, physicalY: location.dy),
         ui.PointerData(change: ui.PointerChange.hover, kind: kind, physicalX: location.dx, physicalY: location.dy),
         ui.PointerData(change: ui.PointerChange.down, kind: kind, physicalX: location.dx, physicalY: location.dy),
-        ui.PointerData(change: ui.PointerChange.move, kind: kind, physicalX: location.dx, physicalY: location.dy),
+        ui.PointerData(change: ui.PointerChange.move, buttons: kSecondaryButton, kind: kind, physicalX: location.dx, physicalY: location.dy),
```

Comment:
This does not feel like an event sequence that the embedding would send in real situation, empty buttons at down then secondary buttons at move. Can you make this a separate test case? Making it as close to what you capture on iPadOS as possible would be preferred.